core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionServicesConfig.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.util.compaction;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.conf.ConfigurationTypeHelper;
+import org.apache.accumulo.core.conf.Property;
+import org.apache.accumulo.core.spi.compaction.CompactionServiceId;
+import org.apache.accumulo.core.spi.compaction.DefaultCompactionPlanner;
+
+import com.google.common.collect.Sets;
+
+public class CompactionServicesConfig {
```

Comment:
Since this class is specific to certain properties, it would be nice to note them in the javadoc. Examples of properties is always helpful too.

Reply:
Do you think something similar to the example input file I have in my comment above ([here](https://github.com/apache/accumulo/pull/2345#issuecomment-999719164)) would be a good example to include in the javadoc? And then just mention that the class is specific to compaction properties? @milleruntime

Reply:
It might be a pain to try to put the json in the javadoc so not sure how feasable that would be. At the very least I think we could mention the property names that it is looking at.

Reply:
Gave it a shot in bffe840 let me know what you think.

Reply:
LGTM